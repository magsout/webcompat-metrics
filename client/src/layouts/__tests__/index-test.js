/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";

import Component from "../";

it("renders Component default correctly", () => {
  Object.defineProperty(window, "matchMedia", {
    value: jest.fn(() => {
      return { matches: false };
    }),
  });
  const tree = renderer.create(<Component />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Component matchMedia return false", () => {
  Object.defineProperty(window, "matchMedia", {
    value: jest.fn(() => {
      return { matches: true };
    }),
  });
  const tree = renderer.create(<Component />).toJSON();
  expect(tree).toMatchSnapshot();
});
