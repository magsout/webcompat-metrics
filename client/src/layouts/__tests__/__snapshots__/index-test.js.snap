// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Component default correctly 1`] = `
<div
  className="component"
>
  <div
    className="component"
  >
    <div
      className="component"
      role="banner"
    >
      <div
        className="menu"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          <span
            className="SVGInline SVGInline--cleaned"
            dangerouslySetInnerHTML={
              Object {
                "__html": "test-file-stub",
              }
            }
          />
        </button>
      </div>
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="link"
        >
          <div
            className="baseline"
          >
            <div>
              Webcompat
            </div>
            <div>
              Metrics
            </div>
          </div>
          <div
            className="logo"
          >
            <span
              className="SVGInline SVGInline--cleaned"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
            />
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className="component isOpen"
  >
    <div
      className="container"
    >
      <nav
        className="links"
        role="navigation"
      >
        <a
          href="/needsdiagnosis"
          onClick={[Function]}
        >
          <div
            className="component"
          >
            <div
              className="icon"
            >
              <span
                className="SVGInline SVGInline--cleaned"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "test-file-stub",
                  }
                }
              />
            </div>
            <div
              className="text"
            >
              NeedsDiagnosis
            </div>
          </div>
        </a>
      </nav>
      <footer
        className="footer"
      >
        <div
          className="socials"
        >
          <a
            className="social"
            href="https://twitter.com/webcompat"
            rel="noopener noreferrer"
            target="_blank"
            title="Twitter webcompat accounts"
          >
            <span
              className="SVGInline SVGInline--cleaned"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
            />
          </a>
          <a
            className="social"
            href="https://github.com/webcompat"
            rel="noopener noreferrer"
            target="_blank"
            title="Github webcompat account"
          >
            <span
              className="SVGInline SVGInline--cleaned"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
            />
          </a>
        </div>
        <div
          className="footerText"
        >
          <span>
            Created and maintained by Webcompat
          </span>
        </div>
      </footer>
    </div>
  </div>
  <main
    className="component"
    role="main"
  />
</div>
`;

exports[`renders Component matchMedia return false 1`] = `
<div
  className="component"
>
  <div
    className="component"
  >
    <div
      className="component"
      role="banner"
    >
      <div
        className="menu"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          <span
            className="SVGInline SVGInline--cleaned"
            dangerouslySetInnerHTML={
              Object {
                "__html": "test-file-stub",
              }
            }
          />
        </button>
      </div>
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="link"
        >
          <div
            className="baseline"
          >
            <div>
              Webcompat
            </div>
            <div>
              Metrics
            </div>
          </div>
          <div
            className="logo"
          >
            <span
              className="SVGInline SVGInline--cleaned"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
            />
          </div>
        </div>
      </a>
    </div>
  </div>
  <div
    className="component "
  >
    <div
      className="container"
    >
      <nav
        className="links"
        role="navigation"
      >
        <a
          href="/needsdiagnosis"
          onClick={[Function]}
        >
          <div
            className="component"
          >
            <div
              className="icon"
            >
              <span
                className="SVGInline SVGInline--cleaned"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "test-file-stub",
                  }
                }
              />
            </div>
            <div
              className="text"
            >
              NeedsDiagnosis
            </div>
          </div>
        </a>
      </nav>
      <footer
        className="footer"
      >
        <div
          className="socials"
        >
          <a
            className="social"
            href="https://twitter.com/webcompat"
            rel="noopener noreferrer"
            target="_blank"
            title="Twitter webcompat accounts"
          >
            <span
              className="SVGInline SVGInline--cleaned"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
            />
          </a>
          <a
            className="social"
            href="https://github.com/webcompat"
            rel="noopener noreferrer"
            target="_blank"
            title="Github webcompat account"
          >
            <span
              className="SVGInline SVGInline--cleaned"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "test-file-stub",
                }
              }
            />
          </a>
        </div>
        <div
          className="footerText"
        >
          <span>
            Created and maintained by Webcompat
          </span>
        </div>
      </footer>
    </div>
  </div>
  <main
    className="component"
    role="main"
  />
</div>
`;
